board/safety/safety_gm.h

```diff
@@ -85,19 +86,16 @@ static int gm_rx_hook(CANPacket_t *to_push) {
     if (addr == 241) {
       // Brake pedal's potentiometer returns near-zero reading
       // even when pedal is not pressed
-      brake_pressed = GET_BYTE(to_push, 1) >= 10U;
+      brake_pressed = (GET_BYTE(to_push, 1) >= 10U) || (gm_regen_braking != 0U);
     }
 
     if (addr == 452) {
       gas_pressed = GET_BYTE(to_push, 5) != 0U;
     }
 
-    // exit controls on regen paddle
+    // Regen braking is braking
     if (addr == 189) {
-      bool regen = GET_BYTE(to_push, 0) & 0x20U;
-      if (regen) {
-        controls_allowed = 0;
-      }
+      gm_regen_braking = GET_BYTE(to_push, 0) >> 4;
```

Comment:
you can just make a global regen_braking and add it to the check in generic_rx_checks